File: ReturnTracker/Utilities/CurrencyFormatter.swift
import Foundation

enum CurrencyFormatter {
    private static let koreanWon: NumberFormatter = {
        let formatter = NumberFormatter()
        formatter.numberStyle = .currency
        formatter.currencyCode = "KRW"
        formatter.maximumFractionDigits = 0
        return formatter
    }()

    static func won(_ amount: Decimal) -> String {
        koreanWon.string(from: amount as NSDecimalNumber) ?? "\(amount)원"
    }

    static func decimal(from input: String) -> Decimal? {
        let digits = input.filter(\.isNumber)
        guard !digits.isEmpty else { return nil }
        return Decimal(string: digits)
    }
}

File: android/app/src/main/java/com/returntracker/android/domain/CurrencyFormatter.kt
package com.returntracker.android.domain

import java.text.NumberFormat
import java.util.Currency
import java.util.Locale

object CurrencyFormatter {
    private val koreanWon = NumberFormat.getCurrencyInstance(Locale.KOREA).apply {
        currency = Currency.getInstance("KRW")
        maximumFractionDigits = 0
    }

    fun won(amount: Long): String = koreanWon.format(amount)

    fun parseWon(input: String): Long? {
        val digits = input.filter(Char::isDigit)
        return digits.takeIf(String::isNotEmpty)?.toLongOrNull()
    }
}
